upd for description in infoObject

OlgaKartashevaRBK opened the pull request:
part of the changes within  [BUSINESS-306](https://rbkmoney.atlassian.net/browse/BUSINESS-306) task

![Снимок экрана 2021-09-14 в 14 16 16](https://user-images.githubusercontent.com/80775150/133248390-07672b5c-868a-4a63-be86-95cfab7e1fd3.png)
![Снимок экрана 2021-09-14 в 14 17 01](https://user-images.githubusercontent.com/80775150/133248500-b36a59be-45a8-401c-b5a4-4b103a595fd5.png)
![Снимок экрана 2021-09-14 в 14 55 54](https://user-images.githubusercontent.com/80775150/133253607-01b46f96-39ce-4ee6-96e7-7810ec4da2f5.png)

Commit:
upd for description in infoObject

tolkonepiu commented on spec/swagger.yaml:
```diff
@@ -1,69 +1,87 @@
+    ### Формат дат
 
-    Платформа принимает значения date-time в формате, описанном в RFC 3339, например:
+    [Платформа](https://developer.rbk.money/docs/payments/overview/) принимает и возвращает значения DateTime в формате, описанном в [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339):
```
```suggestion
    [Платформа](https://developer.rbk.money/docs/payments/overview/) принимает и возвращает значения date-time в формате, описанном в [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339):
```

Это больше указание, какой именно из форматов в рамках RFC 3339 используется:
<img width="575" alt="Screenshot 2021-09-14 at 14 26 09" src="https://user-images.githubusercontent.com/5084395/133249334-8151855b-2933-4462-9de1-7d49b879ad5c.png">

tolkonepiu commented on spec/swagger.yaml:
```diff
@@ -1,69 +1,87 @@
-    определяющий максимальное время ожидания завершения операции по запросу.
-    Данная отсечка передается в соответствующем заголовке HTTP-запроса:
+    При любом обращении к API в заголовке соответствующего запроса можно передать параметр отсечки по времени,
+    определяющий максимальное время ожидания завершения операции по запросу:
```
Про http заголовок я может быть и оставил

OlgaKartashevaRBK replied:
Если перенесем раздел "Авторизация" (где есть инфо: `Взаимодействие между [мерчантом](https://developer.rbk.money/docs/payments/overview/#lk) и [платформой](https://developer.rbk.money/docs/payments/overview/) осуществляется по защищенному протоколу (HTTPS)`) выше (перед "Детали взаимодействия")
ок ли будет в дальнейшем не уточнять по тексту, что речь о HTTP-запросах?

tolkonepiu replied:
Тут вопрос скорее не о http-запросах, а о указании параметров в этих запросах. Не совсем очевидно по тексту, что этот параметр нужно указывать в заголовках.

keynslug replied:
Тут соглашусь с @tolkonepiu: раньше было понятнее, что речь шла про один конкретный заголовок запроса (их может быть много).

OlgaKartashevaRBK replied:
Так ок?

При любом обращении к API в заголовке `X-Request-Deadline` соответствующего запроса можно передать параметр отсечки по времени, определяющий максимальное время ожидания завершения операции по запросу:

Если да, тогда и выше уточню

При любом обращении к API в заголовке `X-Request-ID` соответствующего запроса необходимо передать его уникальный идентификатор:

keynslug replied:
Вроде ок.

tolkonepiu commented on spec/swagger.yaml:
```diff
@@ -73,41 +91,7 @@ schemes:
-
-      Обратите внимание, что API-ключ должен использоваться исключительно для межсерверного взаимодействия.
-
-      Для авторизации запросов, выполняемых из контекста браузера плательщика (например, токенизация платежных данных методом [createPaymentResource](https://developer.rbk.money/api/#operation/createPaymentResource)) используйте полученный при вызове [createInvoice](https://developer.rbk.money/api/#operation/createInvoice) `invoiceAccessToken`, либо создайте его напрямую, вызвав метод [createInvo ...
```
А эта часть дропается или куда-то переместилась?

OlgaKartashevaRBK replied:
<img width="1173" alt="image" src="https://user-images.githubusercontent.com/80775150/133405255-803e632b-0159-4c53-9717-5746f6803339.png">

<img width="731" alt="image" src="https://user-images.githubusercontent.com/80775150/133402688-9c883228-6250-4343-b756-5268d0ee1d9a.png">

<img width="759" alt="image" src="https://user-images.githubusercontent.com/80775150/133402732-8186e183-0cb9-4575-ab30-7445799ee5b4.png">

При необходимости могу внести какую-либо доп информацию в статью https://developer.rbk.money/docs/payments/overview/#api

tolkonepiu replied:
Ага, понял

tolkonepiu commented on spec/swagger.yaml:
```diff
@@ -1,69 +1,87 @@
-    меньше **3 секунд** и превышало **1 минуту**.
+      | Формат  | Единицы измерения | Рекомендуемые значения  | Пример 
+      | --         | --          | --         | --         
+      | Формат из раздела "Формат дат" или относительные величины    | `ms` — миллисекунды; `s` — секунды; `m` — минуты    | `3s` < `Request-Deadline` < `1m`    | `150000ms`, `540s`, `3.5m`    | 
```
Кажется стало менее понятно, что дату можно указать и в формате date-time RFC 3339 и в относительных величинах. @keynslug что думаешь?

keynslug replied:
Есть такое, согласен. Ну и формат подачи в виде таблицы из одной строки мне кажется немного сомнительным. Чуть более подробным абзацем с текстом было кажется было бы понятнее. <sup>(если мы не хотим конечно подробно специфицировать формат «относительных дедлайнов», тогда я бы это в отдельном раздельчике сделал) </sup>

OlgaKartashevaRBK replied:
вместо 
`Формат из раздела "Формат дат" или относительные величины"`
укажу, что дату можно указать и в формате date-time RFC 3339, и в относительных величинах

Ок, если таблица не привнесла ясности - напишу текстом

Commit:
upd for securityDefinitions

Validation errors:
[
  {
    "code": "UNRESOLVABLE_REFERENCE",
    "message": "Security definition could not be resolved: bearer",
    "path": [
      "security",
      "0",
      "bearer"
    ]
  }
]

OlgaKartashevaRBK committed:
Update spec/swagger.yaml

Co-authored-by: Pavel Popov <tolkonepiu@users.noreply.github.com>

tolkonepiu reviewed (commented):
<img width="316" alt="Screenshot 2021-09-15 at 12 32 01" src="https://user-images.githubusercontent.com/5084395/133408807-15731169-c0c1-4a90-a08c-f35f512bea8e.png">
Эти файлики надо будет удалить еще

keynslug commented on spec/swagger.yaml:
```diff
@@ -1,69 +1,66 @@
+    ### Формат дат
 
-    Платформа принимает значения date-time в формате, описанном в RFC 3339, например:
+    [Платформа](https://developer.rbk.money/docs/payments/overview/) принимает и возвращает значения date-time в формате, описанном в [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339):
```
Быть может тогда как-то так?
```suggestion
    [Платформа](https://developer.rbk.money/docs/payments/overview/) принимает и возвращает значения отметок во времени в формате `date-time`, описанном в [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339):
```

OlgaKartashevaRBK replied:
Может так?

[Платформа](https://developer.rbk.money/docs/payments/overview/) принимает и возвращает значение отметки времени в формате `date-time`, описанном в [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339):

keynslug replied:
Нет особых возражений, кроме разве что единственного числа.

OlgaKartashevaRBK replied:
с чем связано возражение про единственное число?

keynslug replied:
С тем, что это воспринимается так, словно мы возвращаем _только_ отметку времени и больше ничего.

OlgaKartashevaRBK replied:
ок

[Платформа](https://developer.rbk.money/docs/payments/overview/) принимает и возвращает значения отметок времени в формате `date-time`, описанном в [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339):

keynslug commented on spec/swagger.yaml:
```diff
@@ -1,69 +1,66 @@
   termsOfService: 'https://rbk.money/'
   contact:
-    name: RBKmoney support team
+    name: RBK.money support team
```
Мож перевести на русский тогда?

OlgaKartashevaRBK replied:
Можно тогда просто:

Служба поддержки RBK.money

но в целом в рамках этого PR можно не трогать, там пока все не переведено)
<img width="669" alt="image" src="https://user-images.githubusercontent.com/80775150/133443247-143c2c23-e0e3-4a4f-b263-669212e492e8.png">

keynslug commented on spec/swagger.yaml:
```diff
@@ -73,41 +70,31 @@ schemes:
   - application/json; charset=utf-8
 produces:
-  - application/json; charset=utf-8
+  - application/json; charset=utf-8  
```
Лишнее изменение.

keynslug commented on spec/swagger.yaml:
```diff
@@ -368,7 +355,7 @@ parameters:
       required: true
       type: string
-tags:
+tags:     
```
Лишнее изменение.

keynslug commented on spec/swagger.yaml:
````diff
@@ -73,41 +70,31 @@ schemes:
+      Схема авторизации - [Bearer](https://tools.ietf.org/html/rfc6750).</br>
+      Bearer-токен следует передать в заголовке запроса в поле `Authorization`:
+    
+        ```shell
+        --header "Authorization: Bearer MjMyNDQxMjM6NDUzRmRnZDQ0M*******"
+        ```
````
Мне кажется тут не место для linux shell / curl специфики:
```suggestion
        ```
        Authorization: Bearer MjMyNDQxMjM6NDUzRmRnZDQ0M*******
        ```
```

keynslug commented on spec/swagger.yaml:
```diff
@@ -73,41 +70,31 @@ schemes:
+     type: apiKey
+     name: Authorization
+     in: header
+     description: >
```
Быть может сделать
```suggestion
     description: |
```
и удалить все эти излишние `<br/>`?

OlgaKartashevaRBK replied:
ок

keynslug commented on spec/swagger.yaml:
```diff
@@ -73,41 +70,31 @@ schemes:
+      HTTP-запросы по нешифрованному каналу не поддерживаются.</br></br>
+    
+      Схема авторизации - [Bearer](https://tools.ietf.org/html/rfc6750).</br>
+      Bearer-токен следует передать в заголовке запроса в поле `Authorization`:
```
Речь всё же мне кажется про API ключ должна идти. Мы так везде называем основное средство авторизации. Bearer − это грубо говоря «носитель некого токена», поэтому bearer-token звучит криво.

OlgaKartashevaRBK replied:
Предлагаешь убрать инфо о том,  что принцип авторизации https://tools.ietf.org/html/rfc6750?

Если да, можно просто заменить инфо вот этой строчкой 
<img width="1074" alt="image" src="https://user-images.githubusercontent.com/80775150/133447074-8cb4088d-0f77-4e5d-9d2c-4d4b683a9e75.png">

и получится

Взаимодействие между [мерчантом](https://developer.rbk.money/docs/payments/overview/#lk)
и [платформой](https://developer.rbk.money/docs/payments/overview/) осуществляется по защищенному протоколу (HTTPS).
HTTP-запросы по нешифрованному каналу не поддерживаются.

Ознакомиться с деталями авторизации запросов к API можно в [данной](https://developer.rbk.money/docs/payments/overview/#api) статье.

[Посмотреть](https://help.rbkmoney.com/lk/lk/#apiKey) ваш Bearer-токен (API-ключ) можно
в [личном кабинете](https://developer.rbk.money/docs/payments/overview/#lk).

и т.д.

keynslug replied:
> Посмотреть ваш Bearer-токен (API-ключ) можно

Я пытаюсь сказать, что мы не оперируем понятием Bearer-токен. Поэтому я бы написал просто:
```
Посмотреть ваш API-ключ можно...
```

Или даже так, чтобы futureproof было:
```
Выпустить / создать ваш личный API-ключ вы можете...
```

OlgaKartashevaRBK replied:
Они его не выпускают сами, не очень понимаю чем предложенное не ок:

Ознакомиться с деталями авторизации запросов к API можно в [данной](https://developer.rbk.money/docs/payments/overview/#api) статье.

[Посмотреть](https://help.rbkmoney.com/lk/lk/#apiKey) ваш API-ключ можно в [личном кабинете](https://developer.rbk.money/docs/payments/overview/#lk).

keynslug replied:
> Они его не выпускают сами, не очень понимаю чем предложенное не ок:

Я это и предложил.

keynslug commented on spec/swagger.yaml:
```diff
@@ -73,41 +70,31 @@ schemes:
+      HTTP-запросы по нешифрованному каналу не поддерживаются.</br></br>
+    
+      Схема авторизации - [Bearer](https://tools.ietf.org/html/rfc6750).</br>
+      Bearer-токен следует передать в заголовке запроса в поле `Authorization`:
```
`Authorization` это и есть один из заголовков HTTP-запроса. Поэтому корректней бы звучало `в заголовке Authorization`.

OlgaKartashevaRBK replied:
ок, напишу

... следует передать в заголовке `Authorization`:

keynslug commented on spec/swagger.yaml:
```diff
@@ -73,41 +70,31 @@ schemes:
 
-      Помните, что вы никому не должны передавать ваш API ключ!
+      [Посмотреть](https://help.rbkmoney.com/lk/lk/#apiKey) ваш Bearer-токен (API-ключ) можно
+      в [личном кабинете](https://developer.rbk.money/docs/payments/overview/#lk).</br>
```
А почему не на ЛК просто ссылка?

OlgaKartashevaRBK replied:
потому что там рассказывается, что это, а не просто прямая ссылка на сам UI
+вот здесь [Посмотреть](https://help.rbkmoney.com/lk/lk/#apiKey) и так понятно, куда зайти.

Но если считаешь важным, могу https://developer.rbk.money/docs/payments/overview/#lk заменить на https://dashboard.rbk.money

keynslug replied:
> Но если считаешь важным, могу https://developer.rbk.money/docs/payments/overview/#lk заменить на https://dashboard.rbk.money

Про это и речь. Просто непонятно: ссылка назывется «личный кабинет», а ведёт в доку почему-то.

OlgaKartashevaRBK replied:
вот почему 
<img width="768" alt="image" src="https://user-images.githubusercontent.com/80775150/133459363-b0a73be2-c1aa-478a-a2b7-fc226483512b.png">

но ок, я заменю

OlgaKartashevaRBK replied:
Проговорили с @keynslug , что в итоге данный момент не является принципиальным и решение остается на мое усмотрение

keynslug commented on spec/swagger.yaml:
```diff
@@ -73,41 +70,31 @@ schemes:
+      
+      Ключи не разделяются на тестовые и боевые. Для проведения тестовых транзакций используйте идентификатор [тестового магазина](https://help.rbkmoney.com/lk/lk/#testManagement).</br></br>
+      
+      **Важно! Не передавайте Bearer-токен третьим лицам!**
```
И тут про API-ключ мне кажется лучше написать.

OlgaKartashevaRBK replied:
ок, будет 

**Важно! Не передавайте API-ключ третьим лицам!**

OlgaKartashevaRBK committed:
Update spec/swagger.yaml

Co-authored-by: Andrew Mayorov <a.mayorov@rbkmoney.com>

Commit:
remove hidden files

Commit:
upd for date-time&Request-ID&Request-Deadline&Auth

keynslug commented on .gitignore:
```diff
@@ -60,3 +60,5 @@ package-lock.json
 .idea/gradle.xml
 .idea/libraries
+.DS_Store
+spec/.DS_Store
```
Это правило лишнее по идее при наличии предыдущего.

OlgaKartashevaRBK replied:
От Андрея: "я только про вторую строчку из двух. Мне не сильно помешает, если обе останутся."

keynslug commented on spec/swagger.yaml:
```diff
@@ -46,18 +46,20 @@ info:
-      | Формат из раздела "Формат дат" или относительные величины    | `ms` — миллисекунды; `s` — секунды; `m` — минуты    | `3s` < `Request-Deadline` < `1m`    | `150000ms`, `540s`, `3.5m`    | 
+    `X-Request-Deadline` может:
+    
+    * задаваться в формате `date-time` (см. [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339)) или в относительных величинах;
+    * измеряться в миллисекундах (`150000ms`), секундах (`540s`), минутах (`3.5m`).
```
Тогда уж быть может
```suggestion
    * задаваться в формате `date-time` (см. [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339));
    * задаваться в относительных величинах, в миллисекундах (`150000ms`), секундах (`540s`) или минутах (`3.5m`).
```
?

OlgaKartashevaRBK committed:
Update spec/swagger.yaml

Co-authored-by: Andrew Mayorov <a.mayorov@rbkmoney.com>

OlgaKartashevaRBK commented:
@keynslug @tolkonepiu подскажите плз, вам в итоге ок?

OlgaKartashevaRBK commented:
@rbkmoney/appsec без тебя не могу)
